### PR TITLE
v1 wip

### PR DESCRIPTION
**Pull request by mxcl**

### CODE REVIEW

File: src/utils/Path.ts [hunk 1]
```diff
@@ -42,14 +44,42 @@ export default class Path {
...
-      if (this.string != "/") while (this.string.endsWith("/")) {
-        this.string = this.string.slice(0, -1)
+    }
+
+    if (Deno.build.os == 'windows') {
+      if (!input.match(/^[a-zA-Z]:/)) {
+        if (!input.startsWith("/") && !input.startsWith("\\")) {
+          throw new Error(`invalid absolute path: ${input}`)
+        }
+        input = `C:\\${input}`
```
Comment (jhheider) [inline src/utils/Path.ts:59]: shouldn't this more correctly be from PWD's drive?
```shell
> j:\
> cd this\is\my\test\dir
> pkgx bash \my\test\script\is\on\j:\test.bash
```?

Comment (mxcl) [reply src/utils/Path.ts:59]: Added a comment or two

Comment (jhheider) [reply src/utils/Path.ts:67]: > terminal users are in a non existent directory more often than one might think

this is surprisingly true.

**Comment by mxcl** [comment]
2 remaining issues.

Canot stub fetch in `dnt` mode (ie. the deno dnt module that transpiles us into a valid node module). We worked around this before by just fetching from dist.pkgx.dev but that's no good now (and it was filthy before) since we don't have any packages for windows yet.

Secondly for some reason dnt is compiling koffi for windows and this fails. It shouldn't need compiling since it bundles precompiled blobs with itself. So I don't get it.

**Comment by mxcl** [comment]
All works but I couldn't get LockFileEx to work with ffi. So swapped out for proper-lockfile. This is not acceptable for v1.

**Comment by tannevaled** [comment]
For posterity: opened **pkgxdev/brewkit#346** to pick the Windows-native thread back up from the *build-side* angle (libpkgx#48 covered the runtime). RFC sketches cross-compile-from-Linux via llvm-mingw + a `jq` pilot to break the chicken-and-egg of "no Windows packages yet to test the runtime against".

Cross-linking so anyone tracing the Windows-native history through libpkgx finds the new conversation.

**Pull request merged**